src/wepy/hdf5.py

```diff
@@ -5102,40 +5120,11 @@ def compute_observable(self, func, fields, args,
         # if we are saving this to the trajectories observables add it as a dataset
         if save_to_hdf5:
 
-            for idx_tup, traj_results in zip(result_idxs, results):
-
-                run_idx, traj_idx = idx_tup
-
-                logging.info("Saving run {} traj {} observables/{}".format(
-                    run_idx, traj_idx, field_name))
-
-                # try to get the observables group or make it if it doesn't exist
-                try:
-                    obs_grp = self.traj(run_idx, traj_idx)[OBSERVABLES]
-                except KeyError:
-
-                    logging.info("Group uninitialized. Initializing.")
-
-                    obs_grp = self.traj(run_idx, traj_idx).create_group(OBSERVABLES)
-
-                # try to create the dataset
-                try:
-                    obs_grp.create_dataset(field_name, data=obs_features)
-                # if it fails we either overwrite or raise an error
-                except RuntimeError:
-                    # if we are in a permissive write mode we delete the
-                    # old dataset and add the new one, overwriting old data
-                    if self.mode in ['w', 'w-', 'x', 'r+']:
-
-                        logging.info("Dataset already present. Overwriting.")
-
-                        del obs_grp[field_name]
-                        obs_grp.create_dataset(field_name, data=obs_features)
```

Comment:
This was using the local variable to the loop and not the end `results` variable